web/Services/PostcodeApiService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using BricksAndHearts.Database;
+using Newtonsoft.Json;
+
+namespace BricksAndHearts.Services;
+
+public interface IPostcodeApiService
+{
+    public Task<string> MakeApiRequestToPostcodeApi(string postalCode);
+    public PostcodeioResponseModel TurnResponseBodyToModel(string responseBody);
+}    
+
+
+public class PostcodeApiService : IPostcodeApiService
+{
+    private readonly HttpClient _client = new();
+    private readonly ILogger<PostcodeApiService> _logger;
+    private readonly BricksAndHeartsDbContext _dbContext;
+
+    public PostcodeApiService(ILogger<PostcodeApiService> logger, BricksAndHeartsDbContext dbContext, HttpClient? client = null)
+    {
+        _logger = logger;
+        _dbContext = dbContext;
+        if (client != null)
+        {
+            _client = client;
+        }
+    }
```

Comment:
Do we need to let it be null? It's fine to still pass in `null!` in the tests if we want to.

```suggestion
    public PostcodeApiService(ILogger<PostcodeApiService> logger, BricksAndHeartsDbContext dbContext, HttpClient client)
    {
        _logger = logger;
        _dbContext = dbContext;
        _client = client;
    }
```